### ISSUE TITLE
[Feature Request] Add gutter class

### ISSUE THREAD

**Issue by devistar** [vuetifyjs/vuetify#2601]
### New Functionality
This will allow to add a gutter class between the elements of a parent instead of using padding or margin classes on each elements.

```html
<div id="app">
  <v-app id="inspire">
    <v-content>
      <v-layout row wrap class="gutter-3">
        <v-flex class="md6">
          <v-text-field label="Firstname"></v-text-field>
        </v-flex>
        <v-flex class="md6">
          <v-text-field label="Lastname"></v-text-field>
        </v-flex>
        <v-flex class="md6">
          <v-text-field label="Street"></v-text-field>
        </v-flex>
        <v-flex class="md3">
          <v-text-field label="City"></v-text-field>
        </v-flex>
        <v-flex class="md3">
          <v-text-field label="Country"></v-text-field>
        </v-flex>
        <v-flex class="md12">
          <v-text-field label="Email"></v-text-field>
        </v-flex>
      </v-layout>
    </v-content>
  </v-app>
</div>
```

```css
.gutter-3 {
  box-sizing: border-box;
  margin-left: -16px;
  margin-right: -16px;  
}

.gutter-3 > * {  
   padding-left: 16px;
   padding-right: 16px;
}
```

Codepen : https://codepen.io/anon/pen/zPWmKg

### Improvements
 There will only be one class to add on the parent element. It's faster, easier and consistent. 

### Bugs or Edge Cases it Helps Avoid
It avoids a repetitive task.

**Comment by jrast** [comment]
It seems that this can be achieved using a wrapper `v-container` with `grid-list-{size}` set. However, I don't understand why this is required. It would be nice if it can be solved using a class.

**Comment by johnleider** [comment]
That is a class, and the solution. Grid-lists are what we use for setting the gutter between `v-flex` elements.

`<v-container grid-list-md>` is the same as `<v-container class="grid-list-md">` which is the same as `<div class="container grid-list-md">`

**Comment by ropez** [comment]
@johnleider I came here basically looking for the same thing as the OP.

I read your answer, but I'm struggling to see why the possibility to add gutter is coupled to the `container` class, which also adds padding. What if I only want gutter, and no padding?

For instance, one place where I would need such a thing, is where I have a main container (where I _do_ want the padding), and inside I have a vertical (column) layout. Inside of that I have a row layout where I need gutter. I've tried thing like adding `grid-list-md` to `v-layout`, but it doesn't work:

```html
    <v-container>
      <v-layout column>
        <v-flex>
          First row
        </v-flex>
        <v-layout grid-list-md>   <!-- does nothing -->
          <v-flex sm6>
            First column
          </v-flex>
          <v-flex sm6>
            Second column
          </v-flex>
        </v-layout>
      </v-layout>
    </v-container>
```

If I just wrap the inner `v-layout` in a `v-container`, I get a huge amount of padding, so I need to use a workaround by wrapping the inner layout with something like this:

```html
  <v-container grid-list-md class="pa-0">
```
